### PR TITLE
Logback appender

### PR DESCRIPTION
**Pull request by lehphyro**
As promised in https://github.com/bugsnag/bugsnag-java/pull/77, here's a logback appender that supports configuration of most bugsnag client library properties and takes values from MDC for further customization of reports.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -27,12 +27,14 @@ repositories {
...
     compile 'org.slf4j:slf4j-api:1.7.25'
+    compileOnly 'ch.qos.logback:logback-classic:1.2.3'
     compileOnly 'javax.servlet:javax.servlet-api:3.1.0'
 
+    testCompile 'ch.qos.logback:logback-classic:1.2.3'
     testCompile 'junit:junit:4.12'
-    testCompile 'org.slf4j:slf4j-jdk14:1.7.25'
     testCompile 'javax.servlet:javax.servlet-api:3.1.0'
     testCompile 'org.mockito:mockito-core:2.10.0'
+    testRuntime 'org.codehaus.janino:janino:3.0.8'
```
Comment (fractalwrench) [inline build.gradle]: Is this dependency necessary?

Comment (lehphyro) [reply build.gradle]: It's used to make conditional logback configuration work, it's true it's not relevant for the implementation though.

**Comment by lehphyro** [comment]
Do you have a date for when this is going to be in a release?

**Comment by kattrali** [comment]
@lehphyro Not quite yet. There's a bit more work that we need to do on our end around the product parts of releasing this, like the documentation. There might be a few changes that we need to make as a part of that, though we can take it over the finish line from here as this is a great start. There should be an update soon though.

**Comment by fractalwrench** [comment]
Apologies for leaving this for so long. We plan on releasing a Logback appender as part of #90, along with additional support for the Spring Framework.

**Pull request closed without merging**